### PR TITLE
Converted python code to pylint compliant.

### PR DESCRIPTION
**Pull request by ghost**
Updating the code to make it pylint compliant.

### CODE REVIEW

File: j2lint/linter/error.py [hunk 1]
```diff
@@ -21,24 +23,22 @@ def __repr__(self, verbose=False):
...
                                 "severity": self.rule.severity
                                 })
         else:
             if not settings.verbose:
-                formatstr = u"{2}:{3} {5} ({6})"
+                formatstr = "{2}:{3} {5} ({6})"
             else:
-                formatstr = u"Linting rule: {0}\nRule description: {1}\nError line: {2}:{3} {4}\nError message: {5}\n"
+                formatstr = "Linting rule: {0}\nRule description: " \
+                            "{1}\nError line: {2}:{3} {4}\nError message: {5}\n"
```
Comment (gmuloc) [inline j2lint/linter/error.py]: Do not use `\` to break line - cf PEP8: https://peps.python.org/pep-0008/#maximum-line-length
```suggestion
                formatstr = ("Linting rule: {0}\nRule description: "
                             "{1}\nError line: {2}:{3} {4}\nError message: {5}\n")
```

File: j2lint/linter/indenter/node.py [hunk 2]
```diff
@@ -74,20 +76,21 @@ def check_indent_level(self, result, node):
             node (Node): Node object for which to check the level is correct
         """
         actual = node.statement.begin
-        if len(jinja_node_stack) and jinja_node_stack[0].statement.start_delimeter in JINJA_START_DELIMETERS:
-            BLOCK_START_INDENT = 1
-        elif node.expected_indent == 0 and node.statement.start_delimeter in JINJA_START_DELIMETERS:
-            BLOCK_START_INDENT = 1
+        if jinja_node_stack and jinja_node_stack[0].statement.start_delimeter \
```
Comment (gmuloc) [inline j2lint/linter/indenter/node.py]: Same comment - do not use `\` to break line - cf PEP8

File: j2lint/linter/rule.py [hunk 3]
```diff
@@ -48,26 +54,22 @@ def checklines(self, file, text):
         """
         errors = []
 
-        if not self.check:
-            logger.debug("Check line rule does not exist for {}".format(
-                __class__.__name__))
-            return errors
-
         if not self.is_valid_language(file):
```
Comment (gmuloc) [inline j2lint/linter/rule.py:62]: so `is_valid_language` is made a class method but then used on the instance. Does it really need to be a class method then?

File: j2lint/linter/rule.py [hunk 4]
```diff
@@ -14,15 +13,22 @@ class Rule:
...
         return self.id + ": " + self.description
 
-    def is_valid_language(self, file):
+    @classmethod
+    def checktext(cls, file, text):
+        """ class method for checktext """
+
+    @classmethod
+    def check(cls, line):
+        """ class method for checklines """
```
Comment (gmuloc) [inline j2lint/linter/rule.py]: is it for checklines?

File: j2lint/linter/rule.py [hunk 5]
```diff
@@ -14,15 +13,22 @@ class Rule:
...
 
     def __repr__(self):
         return self.id + ": " + self.description
 
-    def is_valid_language(self, file):
+    @classmethod
+    def checktext(cls, file, text):
+        """ class method for checktext """
+
+    @classmethod
```
Comment (gmuloc) [inline j2lint/linter/rule.py]: Why make these class methods? they are overwritten as method in the various rule files? Please also document in the docstring that you expect a Rule to return a certain format when implementing this method.

File: j2lint/linter/rule.py [hunk 6]
```diff
@@ -80,20 +82,16 @@ def checkfulltext(self, file, text):
...
-                __class__.__name__))
-            return errors
-
         if not self.is_valid_language(file):
             logger.debug(
-                "Skipping file {}. Linter does not support linting this file type".format(file))
+                "Skipping file %s. Linter does not support linting this file type", file)
             return errors
 
         results = self.checktext(file, text)
```
Comment (gmuloc) [inline j2lint/linter/rule.py:100]: So now the assumption is that if a child class does not implement checktext we just use this class `checktext` method and return None? - This works but maybe then state it clearly in the checktext doc string.

Even better could be to return the empty list rather than None

File: j2lint/rules/jinja_template_indentation_rule.py [hunk 7]
```diff
@@ -14,10 +12,12 @@ class JinjaTemplateIndentationRule(Rule):
     """
     id = 'S3'
     short_description = 'jinja-statements-indentation'
-    description = "All J2 statements must be indented by 4 more spaces within jinja delimiter. To close a control, end tag must have same indentation level."
+    description = "All J2 statements must be indented by 4 more spaces within jinja delimiter. " \
```
Comment (gmuloc) [inline j2lint/rules/jinja_template_indentation_rule.py]: no `\`

File: j2lint/linter/rule.py [hunk 8]
```diff
@@ -80,20 +82,16 @@ def checkfulltext(self, file, text):
...
 
-        for line, section, message in results:
-            errors.append(LinterError(
-                line, section, file['path'], self, message))
+        if results is not None:
+            for line, section, message in results:
+                errors.append(LinterError(
+                    line, section, file['path'], self, message))
```
Comment (gmuloc) [inline j2lint/linter/rule.py]: if the default `checktext` method returns `[]` you can remove the `if results is not None` line

File: j2lint/rules/jinja_operator_has_spaces_rule.py [hunk 9]
```diff
@@ -1,33 +1,34 @@
...
 
-class JinjaOperatorHasSpaceRule(Rule):
+class JinjaOperatorHasSpacesRule(Rule):
     """Rule class to check if jinja filter has surrounding spaces.
     """
     id = 'S2'
     short_description = 'operator-enclosed-by-spaces'
-    description = "When variables are used in combination with an operator, the operator shall be enclosed by space: '{{ my_value | to_json }}'"
+    description = "When variables are used in combination with an operator, " \
+                  "the operator shall be enclosed by space: '{{ my_value | to_json }}'"
```
Comment (gmuloc) [inline j2lint/rules/jinja_operator_has_spaces_rule.py]:
```suggestion
    description = ("When variables are used in combination with an operator, "
                   "the operator shall be enclosed by space: '{{ my_value | to_json }}'")
```

File: j2lint/rules/jinja_operator_has_spaces_rule.py [hunk 10]
```diff
@@ -1,33 +1,34 @@
...
     operators = ['|', '+', '==']
     regexes = []
     for operator in operators:
         operator = "\\" + operator
         regex = r"({[{|%](.*?)([^ |^}]" + operator + ")(.*?)[}|%]})|({[{|%](.*?)(" + operator + \
-            "[^ |^{])(.*?)[}|%]})|({[{|%](.*?)([^ |^}] \s+" + operator + \
+            r"[^ |^{])(.*?)[}|%]})|({[{|%](.*?)([^ |^}] \s+" + operator + \
             ")(.*?)[}|%]})|({[{|%](.*?)(" + \
-            operator + " \s+[^ |^{])(.*?)[}|%]})"
+            operator + r" \s+[^ |^{])(.*?)[}|%]})"
```
Comment (gmuloc) [inline j2lint/rules/jinja_operator_has_spaces_rule.py]: no `\`

File: j2lint/rules/jinja_operator_has_spaces_rule.py [hunk 11]
```diff
@@ -50,10 +51,12 @@ def check(self, file, line):
                 issues.append(operator)
         if issues:
             if len(issues) > 1:
-                self.description = "The operators %s need to be enclosed by a single space on each side" % (
-                    ", ".join(issues))
+                self.description = f"The operators {(', '.join(issues))} need to be enclosed " \
```
Comment (gmuloc) [inline j2lint/rules/jinja_operator_has_spaces_rule.py]: no `\`

File: j2lint/rules/jinja_template_indentation_rule.py [hunk 12]
```diff
@@ -29,21 +29,25 @@ def checktext(self, file, text):
...
-            except Exception as e:
-                logger.error("Indentation check failed for file %s: Error: %s" %
-                             (file['path'], str(e)))
+            except SyntaxError as error:
+                logger.error("Indentation check failed for file %s: Error: %s",
+                             file['path'], str(error))
             for error in errors:
                 result.append((error[0], error[1], error[2]))
 
+        logger.debug("Check text rule does not exist for %s",
```
Comment (gmuloc) [inline j2lint/rules/jinja_template_indentation_rule.py]: I am not sure this is correct here - we could have an answer right? if line 48 appends some values

File: j2lint/rules/jinja_template_no_tabs_rule.py [hunk 13]
```diff
@@ -16,14 +16,17 @@ class JinjaTemplateNoTabsRule(Rule):
...
 
         Args:
-            file (string): file path
             line (string): a single line from the file
 
         Returns:
             Object: Returns error object if found else None
         """
+
+        logger.debug("Check line rule does not exist for %s",
```
Comment (gmuloc) [inline j2lint/rules/jinja_template_no_tabs_rule.py]: how can we print this log before doing any test? (done on line 32)

File: j2lint/rules/jinja_template_syntax_error_rule.py [hunk 14]
```diff
@@ -28,12 +29,16 @@ def checktext(self, file, text):
...
                 block = template.read()
                 env.parse(block)
-            except jinja2.TemplateSyntaxError as e:
-                result.append((e.lineno, text.split(
-                    "\n")[e.lineno - 1], e.message))
+            except jinja2.TemplateSyntaxError as error:
+                result.append((error.lineno, text.split(
+                    "\n")[error.lineno - 1], error.message))
+
+        logger.debug("Check text rule does not exist for %s",
```
Comment (gmuloc) [inline j2lint/rules/jinja_template_syntax_error_rule.py]: result could have a value so the log is weird here

File: j2lint/rules/jinja_variable_name_format_rule.py [hunk 15]
```diff
@@ -34,4 +35,8 @@ def check(self, file, line):
             if matches:
                 print(matches)
```
Comment (gmuloc) [inline j2lint/rules/jinja_variable_name_format_rule.py]: Hmm I know you did not change this but wondering what the print statement is doing here

File: j2lint/settings.py [hunk 16]
```diff
@@ -1,6 +1,6 @@
 """settings.py - Settings class for jinja2 linter.
 """
-
+# pylint: disable=too-few-public-methods
 
 class Settings:
```
Comment (gmuloc) [inline j2lint/settings.py:6]: Looking at how this class is used throughout the repo I am wondering if it is really useful (we won't refactor it as part of this PR). As far as I can see it only is really used when printing the `LintingError` in the `__repr__` method which is already a bit of an abuse of the `__repr__` method: https://docs.python.org/3/reference/datamodel.html#object.__repr__

File: j2lint/utils.py [hunk 17]
```diff
@@ -25,17 +23,18 @@ def load_plugins(directory):
...
-            logger.debug("Loading plugin {}".format(pluginname))
+            logger.debug("Loading plugin %s", pluginname)
             spec = importlib.util.spec_from_file_location(
                 pluginname, pluginfile)
-            module = importlib.util.module_from_spec(spec)
-            spec.loader.exec_module(module)
-            obj = getattr(module, pluginname)()
-            result.append(obj)
+            if pluginname != "__init__":
+                class_name = ''.join(str(name).capitalize()for name in pluginname.split('_'))
```
Comment (gmuloc) [inline j2lint/utils.py]: Hmm so here we are now assuming that if the user want to use a different directory of rules they will always name the class based on the filename and doing the same process (removing `_` and capitalizing)

Here the goal for us is not to refactor this either but if this is not documented it is quite hard to maintain this. We have a section in the README indicating how to add custom rules maybe worth mentioning it there.

File: j2lint/linter/indenter/node.py [hunk 18]
```diff
@@ -120,9 +123,9 @@ def check_indentation(self, result, lines, line_no=0, indent_level=0):
                     result, lines, line_no + 1, indent_level + INDENT_SHIFT)
                 self.check_indent_level(result, node)
                 continue
-            elif node.tag in END_TAGS:
+            if node.tag in END_TAGS:
```
Comment (gmuloc) [inline j2lint/linter/indenter/node.py:134]: I guess pylint may be complaining but I would still keep the `elif` here to help readability - even if I agree that because we get out of each statement it is ok to have `if`. Did you have a warning?

**Review by gmuloc — CHANGES_REQUESTED** [review]
Very good intiative!

This PR is mixing refactoring code functionality and linting so it is a bit hard to follow.
I cannot guarantee 100% that you are not changing some functionality. Were you able to test on a couple of examples at least?

A couple of points:
* don't use `\` to break lines except in very special occasion (cf PEP8)
* I am wondering if simply you dont want `@staticmethod` rather than `@classmethod` everywhere. I guess you were trying to escape the infamous `method could be a function` pylint warning. Though in this case it makes life a bit less understandable. There is not really a reason for the checktext method to be a classmethod if the only place where we are using the class is in some debug log.
* Sometimes pylint is having strong opinions that can decrease readability

File: j2lint/linter/indenter/node.py (hunk 18 above)
Comment (Shivani-chourasiya) [reply j2lint/linter/indenter/node.py:134]: yes , I got a warning for using elif with continue

File: j2lint/rules/jinja_template_indentation_rule.py (hunk 12 above)
Comment (Shivani-chourasiya) [reply j2lint/rules/jinja_template_indentation_rule.py]: log message was incorrect , I have corrected it now.

File: j2lint/linter/indenter/node.py (hunk 18 above)
Comment (gmuloc) [reply j2lint/linter/indenter/node.py:134]: ok we can keep it that way - I will add an issue to refactor this method at some point

File: j2lint/rules/jinja_statement_has_space_rule.py [hunk 19]
```diff
@@ -27,7 +27,7 @@ def check(self, line):
         Returns:
             Object: Returns error object if found else None
         """
-        logger.debug("Check line rule does not exist for %s",
+        logger.debug("Check text rule does not exist for %s",
```
Comment (gmuloc) [inline j2lint/rules/jinja_statement_has_space_rule.py]: I don't understand why you state `does not exist` - how do you know if does not exist here? the test is done afterin the regex. Am I missing something?

**Review by gmuloc — CHANGES_REQUESTED** [review]
What is the purpose of the log in the class? It looks like it was added only to avoid to trigger `unused import` - more logging is good but some of these log messages do not make sense.

What do you mean with `Check * rule does not exist` - are you saying that the rule was not triggered for this given line? If so it maybe make sense for the method that returns True / False but for the one returning a result or a regex it is not possible to assert before the `return` statement if the `check` exist or not and so the logs seem incorrect.

Two things:
1. It is better to log _when_ a rule is triggered in debug rather than when it is not I would argue
2. If the only purpose of the log message is to cater for `unused-import` I would not add it

File: j2lint/utils.py (hunk 17 above)
Comment (Shivani-chourasiya) [reply j2lint/utils.py]: Added to the readme.

File: j2lint/linter/rule.py (hunk 5 above)
Comment (Shivani-chourasiya) [reply j2lint/linter/rule.py]: I have corrected the docstring. For this , I was getting a warning "checktext/check is not callable" that's why added these class methods.

File: j2lint/linter/rule.py (hunk 6 above)
Comment (Shivani-chourasiya) [reply j2lint/linter/rule.py:100]: if we return an empty list in this , will get another warning for "arguments file and text are not used" that's why returned None.

File: j2lint/rules/jinja_statement_has_space_rule.py (hunk 19 above)
Comment (Shivani-chourasiya) [reply j2lint/rules/jinja_statement_has_space_rule.py]: I had added these logs as some rules are having check method and some are having checktext, but now as we are defining the default methods in rule.py  I think we can remove these logs.

File: README.md [hunk 20]
```diff
@@ -134,8 +134,12 @@ j2lint <path-to-directory-of-templates> --ignore <rule_description1> <rule_desc>
...
 
 1. Create a new rules directory under j2lint folder.
-2. Add custom rule classes which are similar to classes in j2lint/rules directory.
-3. Run the jinja2 linter using --rules-dir option
+2. Add custom rule classes in j2lint/rules directory.
+3. The file name of rules should be in snake_case and the class name should be the PascalCase version of file name.
+   For example- 
+          File name - jinja_operator_has_spaces_rule.py
+          Class name - JinjaOperatorHasSpacesRule
+4. Run the jinja2 linter using --rules-dir option
```
Comment (gmuloc) [inline README.md]: So this here tells to add the rules in the existing folder whereas step 1. is indicating to create the custom folder - my suggestion may need to be adapted as I cam no
```suggestion
2. Add custom rule classes which are similar to classes in j2lint/rules directory:
    The file name of rules should be in snake_case and the class name should be the PascalCase version of file name. For example: 
    * File name - jinja_operator_has_spaces_rule.py
     * Class name - JinjaOperatorHasSpacesRule
3. Run the jinja2 linter using --rules-dir option
```

File: j2lint/cli.py [hunk 21]
```diff
@@ -78,13 +77,33 @@ def sort_issues(issues):
...
+    for file_name in files:
+        runner = Runner(collection, file_name, checked_files)
+        if file_name not in lint_errors:
+            lint_errors[file_name] = []
+        if file_name not in lint_warnings:
+            lint_warnings[file_name] = []
+        j2_errors, j2_warnings = runner.run()
+        lint_errors[file_name].extend(j2_errors)
+        lint_warnings[file_name].extend(j2_warnings)
+    print_linting_result(options, lint_warnings, lint_errors)
```
Comment (gmuloc) [inline j2lint/cli.py]: Lets leave the print part in the `run` function to decorrelate the functionality of `get` from the outformat.

File: j2lint/cli.py [hunk 22]
```diff
@@ -78,13 +77,33 @@ def sort_issues(issues):
...
-            issue.linenumber,
+            issue.line_number,
             issue.rule.id
         )
     )
     return issues
 
 
+def get_linting_issues(file_or_dir_names, options, collection, checked_files):
+    """ checking errors and warnings in files """
```
Comment (gmuloc) [inline j2lint/cli.py]: extend the docstring

File: j2lint/cli.py [hunk 23]
```diff
@@ -175,39 +221,9 @@ def run(args=None):
...
-    elif not total_errors and not total_warnings:
-        print("Linting complete. No problems found.")
-    else:
-        print(f"Jinja2 linting finished with "
-              f"{total_errors} issue(s) and {total_warnings} warning(s)")
+    get_linting_issues(file_or_dir_names, options, collection, checked_files)
 
-    if total_errors:
-        return 2
     return 0
```
Comment (gmuloc) [inline j2lint/cli.py]: You are changing the return value of `run` here - There is no way to return anything than `0` now. You should retrieve the output of one of the method you created and return this code

File: j2lint/rules/jinja_variable_name_format_rule.py (hunk 15 above)
Comment (Shivani-chourasiya) [reply j2lint/rules/jinja_variable_name_format_rule.py]: yes,we can remove this

File: j2lint/linter/rule.py (hunk 5 above)
Comment (Shivani-chourasiya) [reply j2lint/linter/rule.py]: Added disable for this warning and removed class method.

File: j2lint/linter/error.py [hunk 24]
```diff
@@ -1,16 +1,18 @@
 """error.py - Error classes to format the lint errors.
 """
+import json
 from j2lint.settings import settings
 from j2lint.logger import logger
-import json
+
+# pylint: disable=too-few-public-methods,too-many-arguments
```
Comment (gmuloc) [inline j2lint/linter/error.py]: Better to disable too-many-arguments at the method level rather than at the file level

File: j2lint/linter/indenter/node.py [hunk 25]
```diff
@@ -50,7 +51,8 @@ def create_node(self, line, line_no, indent_level=0):
         node.parent = self
         return node
 
-    def create_indentation_error(self, node, message):
+    @classmethod
+    def create_indentation_error(cls, node, message):
```
Comment (gmuloc) [inline j2lint/linter/indenter/node.py]: Why make it a `@classmethod` - you can make it a `@staticmethod` here as yoy don;t need to `cls` argument:

```suggestion
    @staticmethod
    def create_indentation_error(node, message):
```

File: j2lint/linter/indenter/node.py [hunk 26]
```diff
@@ -24,10 +23,12 @@ class Node:
     """
     statement = None
     tag = None
+    parent = None
     node_start = 0
     node_end = 0
     children = []
     expected_indent = 0
+    block_start_indent = 0
```
Comment (gmuloc) [inline j2lint/linter/indenter/node.py]: So I guess this was to remove the global variable `BLOCK_START_INDENT`.

Why not use an `def __init__(self)` constructor for Node rather than adding a class variable? is the indentation really shared across all instance of the Node class?

File: j2lint/linter/indenter/node.py [hunk 27]
```diff
@@ -157,14 +160,14 @@ def check_indentation(self, result, lines, line_no=0, indent_level=0):
...
+                message = f"Unsupported tag '{node.tag}' found"
+                error = self.create_indentation_error(node, message)
+                result.append(error)
+                raise JinjaLinterError(
+                    "Unsupported tag '{node.tag}' found")
+
+            self.children.append(node)
+            line_no = line_no + 1
+            self.check_indent_level(result, node)
+            continue
```
Comment (gmuloc) [inline j2lint/linter/indenter/node.py]: Is it not a useless `continue`?

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: j2lint/cli.py (hunk 21 above)
Comment (Shivani-chourasiya) [reply j2lint/cli.py]: it gives warning for too-many-branches , thats why I have added it in separate function

Comment (gmuloc) [reply j2lint/cli.py]: I am not saying this is not a good idea my point is that you refactor like this for the `run` function:

```
def run()
    [...]
    remove_temporary_file(stdin_filename)
    results = get_linting_result()
    print_liniting_result(results)
    [...]
```

This way you keep separated the `get` and the `print` and it will allow for easier refactoring later

This way you can even use the results from `get_linting_error` to return the same value as it is returning now (cf my next comment in this file)

File: README.md [hunk 28]
```diff
@@ -134,7 +134,10 @@ j2lint <path-to-directory-of-templates> --ignore <rule_description1> <rule_desc>
 ### Adding custom rules
 
 1. Create a new rules directory under j2lint folder.
-2. Add custom rule classes which are similar to classes in j2lint/rules directory.
+2. Add custom rule classes which are similar to classes in j2lint/rules directory:
+    The file name of rules should be in snake_case and the class name should be the PascalCase version of file name. For example: 
+    * File name - jinja_operator_has_spaces_rule.py
+     * Class name - JinjaOperatorHasSpacesRule
```
Comment (gmuloc) [inline README.md]:
```suggestion
    * File name - jinja_operator_has_spaces_rule.py
    * Class name - JinjaOperatorHasSpacesRule
```

my bad here - I messed up the indentation in my suggestion..

File: j2lint/rules/jinja_variable_has_space_rule.py [hunk 29]
```diff
@@ -12,20 +12,21 @@ class JinjaVariableHasSpaceRule(Rule):
     """
     id = 'S1'
     short_description = 'single-space-decorator'
-    description = "A single space shall be added between Jinja2 curly brackets and a variable’s name: '{{ ethernet_interface }}'"
+    description = "A single space shall be added between Jinja2 curly brackets" \
```
Comment (gmuloc) [inline j2lint/rules/jinja_variable_has_space_rule.py]: Remove `\`

File: j2lint/linter/error.py (hunk 24 above)
Comment (gmuloc) [inline j2lint/linter/error.py]:
```suggestion
# pylint: disable=too-few-public-methods
```

File: j2lint/linter/error.py [hunk 30]
```diff
@@ -1,16 +1,18 @@
...
 
 
 class LinterError:
     """Class for lint errors.
     """
 
-    def __init__(self, linenumber, line, filename, rule, message=None):
-        self.linenumber = linenumber
+    def __init__(self, line_number, line, filename, rule, message=None):
+        self.line_number = line_number
```
Comment (gmuloc) [inline j2lint/linter/error.py]:
```suggestion
    def __init__(self, line_number, line, filename, rule, message=None):
        # pylint: disable=too-many-arguments
        self.line_number = line_number
```

File: j2lint/cli.py [hunk 31]
```diff
@@ -145,21 +195,19 @@ def run(args=None):
...
+        rules = f"Jinja2 lint rules\n{collection}\n"
         print(rules)
         logger.debug(rules)
         return 0
 
     # Version of j2lint
-    if options.version:
-        print(f"Jinja2-Linter Version {VERSION}")
-        return 0
+    print_j2lint_version(options)
```
Comment (gmuloc) [inline j2lint/cli.py]: This is changing the behavior here

Before:
if you give the version option then you print the version and exit

After:
you always call the version function but do not exit

Please revert this back and remove the method above. let's limit the refactoring.

File: j2lint/cli.py [hunk 32]
```diff
@@ -78,13 +83,39 @@ def sort_issues(issues):
...
-            issue.linenumber,
+            issue.line_number,
             issue.rule.id
         )
     )
     return issues
 
 
+def get_linting_issues(file_or_dir_names, options, collection, checked_files):
+    """ checking errors and warnings in files and printing the result"""
```
Comment (gmuloc) [inline j2lint/cli.py]:
```suggestion
    """ checking errors and warnings"""
```

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: j2lint/cli.py (hunk 31 above)
Comment (gmuloc) [reply j2lint/cli.py]: This is not resolved, the behavior is changed and it should not. Please change it back to:

```suggestion
    if options.version:
        print(f"Jinja2-Linter Version {VERSION}")
        return 0
```

File: j2lint/cli.py [hunk 33]
```diff
@@ -66,6 +65,12 @@ def create_parser():
     return parser
 
 
+def print_j2lint_version(options):
+    """ Version of j2lint """
+    if options.version:
+        print(f"Jinja2-Linter Version {VERSION}")
+
+
```
Comment (gmuloc) [inline j2lint/cli.py]: Not needed - we can explore this later but for now please don't change the program behavior.

```suggestion
```

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: j2lint/cli.py [hunk 34]
```diff
@@ -164,6 +158,9 @@ def run(args=None):
     Returns:
         int: 0 on success
     """
+
+    # pylint: disable=too-many-branches
```
Comment (gmuloc) [inline j2lint/cli.py]:
```suggestion

    # FIXME - remove this during refactoring
    # pylint: disable=too-many-branches
```

File: j2lint/linter/indenter/node.py [hunk 35]
```diff
@@ -21,15 +21,17 @@
...
-    statement = None
-    tag = None
-    parent = None
-    node_start = 0
-    node_end = 0
-    children = []
-    expected_indent = 0
-    block_start_indent = 0
 
+    # pylint: disable=too-many-instance-attributes
```
Comment (gmuloc) [inline j2lint/linter/indenter/node.py:25]:
```suggestion
    # pylint: disable=too-many-instance-attributes
    # Eight arguments is reasonable in this case
```

File: j2lint/utils.py [hunk 36]
```diff
@@ -25,17 +23,30 @@ def load_plugins(directory):
...
-        pluginname = os.path.basename(pluginfile.replace('.py', ''))
+    for plugin_file in glob.glob(os.path.join(directory, "[A-Za-z_]*.py")):
+        plugin_name = os.path.basename(plugin_file.replace(".py", ""))
         try:
-            logger.debug("Loading plugin {}".format(pluginname))
-            spec = importlib.util.spec_from_file_location(
-                pluginname, pluginfile)
-            module = importlib.util.module_from_spec(spec)
-            spec.loader.exec_module(module)
-            obj = getattr(module, pluginname)()
```
Comment (gmuloc) [inline j2lint/utils.py:38]: Hmm here the code was returning an instance of the Rule wherease we are now returning the Class - though through testing it seems everything is still working. Need to double check this

Comment (gmuloc) [reply j2lint/utils.py:38]: ok this is going with another change in collection where we now instantiate the rule_obj there

**Review by gmuloc — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]
LGTM

**Review by carlbuchmann — APPROVED** [review]
LGTM!
